source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -180,8 +165,9 @@ Cluster::LoadBalancer::chooseHost(Upstream::LoadBalancerContext* context) {
     return nullptr;
   }
 
-  const auto host_it = host_map_->find(host);
-  if (host_it == host_map_->end()) {
+  absl::ReaderMutexLock lock{&cluster_.host_map_lock_};
+  const auto host_it = cluster_.host_map_.find(host);
```

Comment:
perf nit: use braces to reduce lock scope to just the find.

Reply:
In the rest of the function we're using members on `cluster_.host_map` and using `host_it` to access the host info so it's not safe to drop the lock until the end of the function, but I'll add braces around the if/else so that the lock won't be held if down the road we need to do some work at the end of the function.